src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -157,11 +219,13 @@ const TopicFragment = ({
 
   return (
     <View>
-      <RecentSearch
-        shown={isFirstTimeOpen}
-        setSearchText={setSearchText}
-        setIsFirstTimeOpen={setIsFirstTimeOpen}
-      />
+      {!withoutRecent && (
+        <RecentSearch
+          shown={isFirstTimeOpen}
+          setSearchText={setSearchText}
+          setIsFirstTimeOpen={setIsFirstTimeOpen}
+        />
+      )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The rendering of the `RecentSearch` component is now conditional on the `withoutRecent` prop. This change seems to be necessary for the new feature.
```diff
- <RecentSearch
-   shown={isFirstTimeOpen}
-   setSearchText={setSearchText}
-   setIsFirstTimeOpen={setIsFirstTimeOpen}
- />
+ {!withoutRecent && (
+   <RecentSearch
+     shown={isFirstTimeOpen}
+     setSearchText={setSearchText}
+     setIsFirstTimeOpen={setIsFirstTimeOpen}
+   />
+ )}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->